🔖 Release version 0.13.0 (#185)
maykinmedia/open-api-framework: CHANGELOG.rst

Changelog
=========

0.13.0 (2025-09-18)
-------------------

> **💥 Breaking changes**

Author: I tested it in both, and it isn’t. Changed it.

> * Specify optional dependency groups:

Reviewer: ```suggestion
* [#178] Specify optional dependency groups:
```

    * **structlog**: ``structlog``, ``django-structlog``
    * **structlog-celery**: ``structlog``, ``django-structlog[celery]``

**New features**

* Add logging envvars:

    * ``LOG_FORMAT_CONSOLE``
    * ``ENABLE_STRUCTLOG_REQUESTS``

> * Add option to configure whether to use structlog or standard logging (internal setting).

Reviewer: ```suggestion
* [#178] Add option to configure whether to use structlog or standard logging (default: stdlib logging). To enable structlog, downstream projects can do the following in their base settings:

.. code-block:: python

    os.environ["_USE_STRUCTLOG"] = "True"

    from open_api_framework.conf.base import *  # noqa
   
```

> 

Reviewer: A PR that was open for some time with a small bugfix was also merged (commit hash 660551c). Because the commit itself was from some time ago, it's not that easy to find via `git log` or via the github commit history, so generally when making changelogs I do: `git log --oneline 0.12.0 release/0.13.0` (with the latest and new versions), that way I see all commits that will be included in the new release

```suggestion

**Bugfixes/QOL**

* Fix casting DB_PORT "" to int
```

**Maintenance**

> * Pin ```csp``` to 4.0 or higher.

Reviewer: ```suggestion
* Pin ```django-csp``` to 4.0 or higher.

.. warning::
   
   The CSP environment variables function the same as with 3.8, but if any changes are made to CSP settings for downstream projects (like adding extra values to directives via the old 3.8 settings), manual action is needed to make sure it works with 4.0 (`see documentation <https://django-csp.readthedocs.io/en/latest/configuration.html#migrating-from-django-csp-3-8>`_)

   
```
